spring-kafka/src/main/java/org/springframework/kafka/core/reactive/ReactiveKafkaProducerOperations.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.kafka.core.reactive;
+
+import java.util.function.Function;
+
+import org.apache.kafka.clients.producer.Producer;
+import org.reactivestreams.Publisher;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.kafka.sender.KafkaOutbound;
+import reactor.kafka.sender.SenderRecord;
+import reactor.kafka.sender.SenderResult;
+import reactor.kafka.sender.TransactionManager;
+
+
+/**
+ * Interface for executing reactive kafka operations on producer side.
+ *
+ * @param <K> the key type.
+ * @param <V> the value type.
+ *
+ * @author Mark Norkin
+ */
+public interface ReactiveKafkaProducerOperations<K, V> {
+
+	<T> Flux<SenderResult<T>> send(Publisher<? extends SenderRecord<K, V, T>> records);
```

Comment:
We probably need do not expose such an API.
I would say a `ProducerRecord` and `org.springframework.kafka.support.SendResult` should be enough from Spring API perspective.

Reply:
it's a bit unclear what client should do if he/she wants matching the response to its corresponding request with correlation metadata. Or your point is that if needed they could check the headers of SendResult.getProducerRecord() ?

Another concern is that SenderResult caries the exception if records sent unsuccessfully, should we extend SendResult with exception then ? 

Also if exposing reactor-kafka types is a concern there is also `public Flux<ReceiverRecord<K, V>> receive()`, however not exposing it would mean there will be no ability to ack the messages manually

Reply:
I see what you mean.
I'm afraid we can't change `SendResult` because it is tied with the `ListenableFuture` returned from the `KafkaTemplate.send()`.

I don't see yet much cleaner solution, so I won't mind to give it a shot and see how it is attractive for end-users.